### ISSUE TITLE
Bug - Datetime at DefaultValueBinder

### ISSUE THREAD

**Issue by phackwer** [PHPOffice/PHPExcel#543]
Greetings!

Line 86 is buggy since it didn't check if the value of the array is an object or instance of DateTime.

Here is an untested small idea of how to fix it. You should place, before string comparison, the following conditional

elseif ($pValue instance of \DateTime) {
return $pValue->format('Y-m-d H:i:s');
}
elseif (is_object($pValue)) {
return $pValue->__toString();
} elseif ($pValue{0} =....

Please, if you can fix this ASAP...

**Comment by brunowego** [comment]
+1

**Comment by MarkBaker** [comment]
As it stands, it'll break Rich Text objects, and I'm not sure if it needs any special handling for objects without a __toString() method (possibly throw some kind of catchable exception, although that requires a custom error handler to convert a catchable fatal error to an exception)

**Comment by phackwer** [comment]
I agree, but anyone who wants to insert any non-SPL object should be obligated to write a __toString method. It's not a bad habit.

**Comment by phackwer** [comment]
Another thing that would be good is to be able to set date mask as a configuration item. So, I could do something like 

PHPExcel::setDateTimeMask('d/m/Y H:i:s');

**Comment by MarkBaker** [comment]
If you want something that will allow you to define a mask, then you want to use a custom value binder, more like the advanced value binder

**Comment by phackwer** [comment]
Humm... The problem in this case is that I do not have a clue of what will come, since it's a generic export-to-excel function. Anything can come. So, we basically have these two options only.

Also, the __toString could be done by only serializing the object if __toString is not avaliable.

But the DateTime is a constant: we do have this issue and formating is not a problem, since people could set it later on  Excel.

**Comment by phackwer** [comment]
The real problem is having an \Exception throw due the lack of a default solution.
